data_structures/binary_tree/validate_binary_search_tree.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import annotations
+
+# Storing inorder traversal values into this list
+traversed_node_values = []  # type: list[int]
+
+
+class Node:
+    """
+    A Node has data variable and pointers to Nodes to its left and right.
+    """
+
+    def __init__(self, data: int) -> None:
+        self.data = data
+        self.left: Node | None = None
+        self.right: Node | None = None
+
+
+def display(tree: Node | None) -> None:  # In Order traversal of the tree
```

Comment:
It would be better if this returned a string that the caller could `print()` or analyze, etc.